src/shared/sentry/sentryadapter.cpp

```diff
@@ -168,3 +157,34 @@ void SentryAdapter::declineCrashReporting() {
   m_userConsent = UserConsentResult::Forbidden;
   emit userConsentChanged();
 }
+
+void SentryAdapter::captureQMLStacktrace(const char* description) {
+  auto engine = QmlEngineHolder::instance()->engine();
+  auto privateEngine = QQmlEnginePrivate::get(engine);
+  QV4::ExecutionEngine* qv4Engine = privateEngine->v4engine();
+  QVector<QV4::StackFrame> stackTrace = qv4Engine->stackTrace(15);
+  sentry_value_t crumb = sentry_value_new_breadcrumb("info", description);
+  sentry_value_set_by_key(crumb, "category",
+                          sentry_value_new_string("stacktrace"));
+  sentry_value_set_by_key(crumb, "level",
+                          sentry_value_new_string("stacktrace"));
+  sentry_value_t data = sentry_value_new_object();
+  for (int i = 0; i < stackTrace.count(); i++) {
+    const QV4::StackFrame& frame = stackTrace.at(i);
+    sentry_value_t sentry_frame = sentry_value_new_object();
+
+    sentry_value_set_by_key(
+        sentry_frame, "function",
+        sentry_value_new_string(frame.function.toLocal8Bit().constData()));
+    sentry_value_set_by_key(
+        sentry_frame, "source",
+        sentry_value_new_string(frame.source.toLocal8Bit().constData()));
+    sentry_value_set_by_key(sentry_frame, "line",
+                            sentry_value_new_int32(frame.line));
+    sentry_value_set_by_key(sentry_frame, "column",
+                            sentry_value_new_int32(frame.column));
+    sentry_value_set_by_key(data, std::to_string(i).c_str(), sentry_frame);
+  }
+  sentry_value_set_by_key(crumb, "data", data);
+  sentry_add_breadcrumb(crumb);
+}
```

Comment:
Does adding this to the Sentry reports require a quick legal consult? Can the QML stack trace contain more personal information than the current Sentry reports contain?

Reply:
Shoud be fine :)